gtsam_unstable/slam/tests/testLocalOrientedPlane3Factor.cpp

```diff
@@ -0,0 +1,240 @@
+/* ----------------------------------------------------------------------------
+
+ * GTSAM Copyright 2010, Georgia Tech Research Corporation,
+ * Atlanta, Georgia 30332-0415
+ * All Rights Reserved
+ * Authors: Frank Dellaert, et al. (see THANKS for the full author list)
+
+ * See LICENSE for the license information
+
+ * -------------------------------------------------------------------------- */
+
+/*
+ * @file testLocalOrientedPlane3Factor.cpp
+ * @date Feb 12, 2021
+ * @author David Wisth
+ * @brief Tests the LocalOrientedPlane3Factor class
+ */
+
+#include <gtsam_unstable/slam/LocalOrientedPlane3Factor.h>
+
+#include <gtsam/base/numericalDerivative.h>
+#include <gtsam/inference/Symbol.h>
+#include <gtsam/nonlinear/ISAM2.h>
+
+#include <CppUnitLite/TestHarness.h>
+
+#include <boost/bind.hpp>
+
+using namespace gtsam;
+using namespace std;
+
+GTSAM_CONCEPT_TESTABLE_INST(OrientedPlane3)
+GTSAM_CONCEPT_MANIFOLD_INST(OrientedPlane3)
+
+using symbol_shorthand::P;  //< Planes
+using symbol_shorthand::X;  //< Pose3
+
+// *************************************************************************
+TEST(LocalOrientedPlane3Factor, lm_translation_error) {
+  // Tests one pose, two measurements of the landmark that differ in range only.
+  // Normal along -x, 3m away
+  OrientedPlane3 test_lm0(-1.0, 0.0, 0.0, 3.0);
+
+  NonlinearFactorGraph graph;
+
+  // Init pose and prior.  Pose Prior is needed since a single plane measurement
+  // does not fully constrain the pose
+  Pose3 init_pose = Pose3::identity();
+  Pose3 anchor_pose = Pose3::identity();
+  graph.addPrior(X(0), init_pose, noiseModel::Isotropic::Sigma(6, 0.001));
+  graph.addPrior(X(1), anchor_pose, noiseModel::Isotropic::Sigma(6, 0.001));
+
+  // Add two landmark measurements, differing in range
+  Vector4 measurement0(-1.0, 0.0, 0.0, 3.0);
+  Vector4 measurement1(-1.0, 0.0, 0.0, 1.0);
+  LocalOrientedPlane3Factor factor0(
+      measurement0, noiseModel::Isotropic::Sigma(3, 0.1), X(0), X(1), P(0));
+  LocalOrientedPlane3Factor factor1(
+      measurement1, noiseModel::Isotropic::Sigma(3, 0.1), X(0), X(1), P(0));
+  graph.add(factor0);
+  graph.add(factor1);
+
+  // Initial Estimate
+  Values values;
+  values.insert(X(0), init_pose);
+  values.insert(X(1), anchor_pose);
+  values.insert(P(0), test_lm0);
+
+  // Optimize
+  ISAM2 isam2;
+  isam2.update(graph, values);
+  Values result_values = isam2.calculateEstimate();
+  auto optimized_plane_landmark = result_values.at<OrientedPlane3>(P(0));
+
+  // Given two noisy measurements of equal weight, expect result between the two
+  OrientedPlane3 expected_plane_landmark(-1.0, 0.0, 0.0, 2.0);
+  EXPECT(assert_equal(optimized_plane_landmark, expected_plane_landmark));
+}
+
+// *************************************************************************
+/*
+TEST (LocalOrientedPlane3Factor, lm_rotation_error) {
```

Comment:
same, why this is commented out?

Reply:
Both of these unit tests are commented out because they fail, but I can't figure out why - see my comments [here](https://github.com/borglab/gtsam/pull/564#issuecomment-779290861). @ProfFan - any intuition as to why these tests might be failing?

Reply:
@ProfFan is super-busy with paper deadline and has probably no Plane3 experience. I think he is just suggesting you add a comment as to why we commented this out, and add a TODO to figure out the reason later. Then we can land this PR after CI runs.

Reply:
No problem, I just pushed some comments for these commented out unit tests